tests/pixeldataset/test_precomputed_layouts.py

```diff
@@ -313,3 +357,165 @@ def test_aggregate_precomputed_layouts_no_layouts_in_data(self):
         )
 
         assert aggregated_layouts.is_empty
+
+
+class TestGeneratePrecomputedLayoutsForComponents:
+    @pytest.fixture(autouse=True)
+    def mock_pool_executor(self):
+        """Mock the pool executor to avoid running tests in parallel.
+
+        The overhead of running in separate processes is not worth it here.
+        """
+
+        class MockPoolExecutor:
+            def __init__(self):
+                pass
+
+            def imap(self, func, *args, **kwargs):
+                yield from map(func, *args)
+
+            def __enter__(self):
+                return self
+
+            def __exit__(self, *args):
+                pass
+
+        with mock.patch(
+            "pixelator.pixeldataset.precomputed_layouts.get_pool_executor",
+        ) as mock_pool_executor:
+            mock_pool_executor.return_value = MockPoolExecutor()
+            yield mock_pool_executor
+
+    @pytest.fixture(name="pixel_dataset")
+    def pixel_dataset_fixture(self, setup_basic_pixel_dataset):
+        (dataset, *_) = setup_basic_pixel_dataset
+        yield dataset
+
+    def test_generate_precomputed_layouts_for_components_with_all_components(
```

Comment:
It follows the convention of using `test_<name of function under test>_<info about test>`. Hence it becomes a bit repetitive here since components come up in both the function name and the test info.